bokehjs/src/lib/models/plots/range_manager.ts

```diff
@@ -46,6 +46,13 @@ export class RangeManager {
       this._update_ranges_together(range_state)   // apply interval bounds while keeping aspect
     }
     this._update_ranges_individually(range_state, {panning, scrolling, maintain_focus})
+
+    // If windowed auto-ranging is configured for a plot axis, then
+    // this call is needed in order to make sure any data-ranges that
+    // are not explicitly overridden at this point also get re-computed
+    if (this.parent.model.window_axis != null) {
+      this.update_dataranges()
+    }
```

Comment:
Maybe this is not needed given the later change to also do this in `_actual_paint` when `window_axis` is set?  cc @mattpap